app/javascript/bundles/MainApp/components/MainApp.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import AppHeader from './AppHeader';
+import AppContent from './AppContent';
+import AppFooter from './AppFooter';
+import AggregatedFilter from './AggregatedFilter';
+import Filter from './Filter';
+
+export default class MainApp extends React.Component {
+  // Using the webpack EnvironmentPlugin to expose RESTAURANTS_API_URL environment variable
+  static restaurantsUrl = process.env.RESTAURANTS_API_URL+'/restaurants.json';
+
+  static generateInitialFilter() {
```

Comment:
Same as before i think this can be dried up.

Another idea to consider `addFilter` could accept an object, then the "random" string values could have more of a meaning.
```
addFilter({
  order: 3,
  value: "max...",
  title: "...",
  function: ()=> ... 
}).addFilter ....
```

Reply:
It is already a class, nevertheless, I understood there is no need in special id and I'm using the filter field name as index and also put all the fields names to designated class (kind of a enum)